services/svc/batch/examples/create_task_thiserror.rs

```diff
@@ -0,0 +1,86 @@
+/*
+Creates a batch job and task using the data plane APIs
+
+This example shows how to do error handling when you wish
+to match any error from azure_svc_batch crate.
+
+cargo run --package azure_svc_batch --example create_task_thiserror
+*/
+
+use azure_identity::token_credentials::AzureCliCredential;
+use azure_svc_batch::models::{JobAddParameter, PoolInformation, TaskAddParameter};
+use std::sync::Arc;
+
+#[tokio::main]
+async fn main() {
+    match run().await {
+        Ok(_) => (),
+        Err(error) => {
+            print_error_chain(error);
+            std::process::exit(1);
+        }
+    }
+}
+
+fn print_error_chain(error: impl std::error::Error) {
```

Comment:
We may want to add that there is ecosystem support for this. For example [`eyre`](https://crates.io/crates/eyre)

Reply:
I added the `eyre` example code and a note.